File: how-to-solve-coding-problems/59.google-interview-on-your-own.ts
let array1 = [1, 2, 3, 9];
let array2 = [1, 2, 4, 4];

function pairHaveSum(arr: number[], targetSum: number): boolean {
    for (let i = 0; i < arr.length; i++) {
        for (let j = 1; j < arr.length; j++) {
            let sum = arr[i] + arr[j];
            if (sum == targetSum) {
                return true;
            }
        }
    }
    return false;
}
console.log(pairHaveSum(array2, 8));

File: 04.how-to-solve-coding-problems/59.google-interview-on-your-own.dart
main() {
  List<int> array1 = [1, 2, 3, 9];
  List<int> array2 = [1, 2, 4, 4];

  bool pairHaveSum(List<int> arr, int targetSum) {
    for (int i = 0; i < arr.length; i++) {
      for (int j = 1; j < arr.length; j++) {
        int sum = arr[i] + arr[j];
        if (sum == targetSum) {
          return true;
        }
      }
    }
    return false;
  } //O(n^2)

  print(pairHaveSum(array2, 8));
}
